modal/_utils/task_command_router_client.py

```diff
@@ -205,29 +206,38 @@ async def send_request(event: grpclib.events.SendRequest) -> None:
         self._stub = TaskCommandRouterStub(self._channel)
 
     def __del__(self) -> None:
-        """Clean up the client when it's garbage collected."""
-        if self._closed:
-            return
+        """Best-effort cleanup if the caller forgot to close().
 
-        self._jwt_refresh_task.cancel()
+        This object is typically used through synchronicity wrappers, which means this finalizer
+        may run on a different thread than the event loop that owns the channel. Closing the
+        channel is therefore scheduled onto the owning loop using call_soon_threadsafe.
 
+        Use getattr in the event that attributes are not yet initialized or the
+        object is in a half-torn-down state.
+        """
         try:
-            self._channel.close()
+            if getattr(self, "_closed", False):
+                return
+            self._closed = True
+
+            channel = getattr(self, "_channel", None)
+            if channel is None:
+                return
+
+            loop = getattr(self, "_loop", None)
+            if loop is not None and not loop.is_closed():
+                loop.call_soon_threadsafe(channel.close)
         except Exception:
+            # During interpreter shutdown, the loop may already be torn down, which could raise an exception.
```

Comment:
Nit: Wouldn't the loop.is_closed() check catch that? Wonder if we could narrow the scope of this exception catch to prevent silently hiding bugs in *our* code. At least we could have it just surround the call_soon_threadsafe

Reply:
True, narrowed the scope